source/faucet/main.d

```diff
@@ -382,6 +382,14 @@ public class Faucet : FaucetAPI
         auto owned_utxo_rng = this.state.owned_utxos.byKeyValue()
             // do not pick already used UTXOs
             .filter!(pair => pair.key !in this.used_utxos);
+
+        assert(owned_utxo_rng.walkLength >= 0);
```

Comment:
walkLength can be a expensive operation - it is linear and not constant - it walks through the entire list to count the number of elements

can we save down walkLength into a separate variable instead of calling twice?

Reply:
Oh I did not know `walkLength` is expensive.
In this case, your suggestion is good.

Reply:
Updated.

Reply:
> Updated.

you can actually make this better you don't need walkLength at all, cause you need only to know if the length is 0 or 1, so I think you can use the InputRange's empty() and front() functions